lib/lightning_web/live/project_live/index.ex

```diff
@@ -52,9 +52,97 @@ defmodule LightningWeb.ProjectLive.Index do
     |> assign(:users, Lightning.Accounts.list_users())
   end
 
+  @impl true
+  def handle_event(
+        "delete_now",
+        %{"id" => project_id},
+        socket
+      ) do
+    project = Projects.get_project(project_id)
+
+    if can_delete_project(socket.assigns.current_user, project) do
+      Projects.delete_project(project)
+      |> case do
+        {:ok, _} ->
+          {:noreply,
+           socket
+           |> assign(
+             active_menu_item: :projects,
+             projects: Projects.list_projects(),
+             page_title: "Projects"
+           )
+           |> put_flash(:info, "Project deleted successfully")
+           |> push_patch(to: ~p"/settings/projects")}
+
+        {:error, changeset} ->
+          {:noreply,
+           socket
+           |> assign(changeset: changeset)
+           |> put_flash(:error, "Can't delete project")}
+      end
+    else
+      {:noreply,
+       put_flash(
+         socket,
+         :error,
+         "You are not authorized to perform this action."
+       )
+       |> push_patch(to: ~p"/settings/projects")}
+    end
+  end
+
+  def handle_event(
+        "cancel_delete",
+        %{"id" => project_id},
+        socket
+      ) do
+    project = Projects.get_project(project_id)
+
+    case Projects.cancel_scheduled_deletion(project) do
+      {:ok, _project} ->
+        {:noreply,
+         socket
+         |> put_flash(:info, "Canceled project deletion schedule")
+         |> push_patch(to: ~p"/settings/projects")}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:noreply, assign(socket, :scheduled_deletion_changeset, changeset)}
+    end
+  end
+
+  def handle_event(
+        "schedule_delete",
+        %{"id" => project_id},
+        socket
+      ) do
+    project = Projects.get_project(project_id)
+
+    case Projects.schedule_project_deletion(project) do
+      {:ok, _project} ->
+        {:noreply,
+         socket
+         |> put_flash(:info, "Project scheduled for deletion")
+         |> push_patch(to: ~p"/settings/projects")}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
```

Comment:
@stuartc another line

Reply:
Ok I've looked into all of these, they are either very difficult to test because we can't really control the conditions in order to trigger a failure; or they have no validations like the one above this one and technically could never return an error.

I don't think these matches should be removed though.